gmf.BackgroundlayerselectorController is broken

Locally I get the following error when trying to open all three applications (mobile, desktop, desktop_alt):

```
AssertionError: Assertion failed: The dimensions object is required
    at new goog.asserts.AssertionError (http://localhost:3000/@/home/tsauerwein/projects/ngeo/node_modules/closure-…9481f1e57fbba986e2304ac80d4173f94c9c/closure/goog/asserts/asserts.js:62:20)
    at Object.goog.asserts.doAssertFailure_ (http://localhost:3000/@/home/tsauerwein/projects/ngeo/node_modules/closure-…481f1e57fbba986e2304ac80d4173f94c9c/closure/goog/asserts/asserts.js:121:11)
    at Object.goog.asserts.assert [as assert] (http://localhost:3000/@/home/tsauerwein/projects/ngeo/node_modules/closure-…481f1e57fbba986e2304ac80d4173f94c9c/closure/goog/asserts/asserts.js:151:18)
    at new gmf.BackgroundlayerselectorController (http://localhost:3000/@/home/tsauerwein/projects/ngeo/contribs/gmf/src/directives/backgroundlayerselector.js:129:16)
    at Object.invoke (http://localhost:3000/node_modules/angular/angular.js:4718:19)
    at $controllerInit (http://localhost:3000/node_modules/angular/angular.js:10354:34)
    at nodeLinkFn (http://localhost:3000/node_modules/angular/angular.js:9263:34)
    at http://localhost:3000/node_modules/angular/angular.js:9673:13
    at processQueue (http://localhost:3000/node_modules/angular/angular.js:16383:28)
    at http://localhost:3000/node_modules/angular/angular.js:16399:27
```

On the online demo, I only see this error for `mobile` and `desktop`.

--- a/contribs/gmf/apps/desktop/index.html
+++ b/contribs/gmf/apps/desktop/index.html
@@ -170,6 +170,7 @@
               <img src="image/background-layer-button.png" alt="" />
             </button>
             <gmf-backgroundlayerselector
+              gmf-backgroundlayerselector-dimensions="::mainCtrl.dimensions"
               gmf-backgroundlayerselector-map="::mainCtrl.map"
               class="dropdown-menu">
             </gmf-backgroundlayerselector>

--- a/contribs/gmf/apps/desktop_alt/index.html
+++ b/contribs/gmf/apps/desktop_alt/index.html
@@ -170,6 +170,7 @@
               <img src="image/background-layer-button.png" alt="" />
             </button>
             <gmf-backgroundlayerselector
+              gmf-backgroundlayerselector-dimensions="::mainCtrl.dimensions"
               gmf-backgroundlayerselector-map="::mainCtrl.map"
               class="dropdown-menu">
             </gmf-backgroundlayerselector>

--- a/contribs/gmf/apps/mobile/index.html
+++ b/contribs/gmf/apps/mobile/index.html
@@ -88,6 +88,7 @@
         id="background"
         class="slide"
         data-header-title="{{'Background' | translate}}"
+        gmf-backgroundlayerselector-dimensions="::mainCtrl.dimensions"
         gmf-backgroundlayerselector-map="::mainCtrl.map"
         gmf-backgroundlayerselector-select="mainCtrl.hideNav()">
       </gmf-backgroundlayerselector>

--- a/contribs/gmf/examples/backgroundlayerselector.html
+++ b/contribs/gmf/examples/backgroundlayerselector.html
@@ -41,6 +41,7 @@
     <div>
         <span>Select base layer:</span>
         <gmf-backgroundlayerselector
+          gmf-backgroundlayerselector-dimensions="::ctrl.dimensions"
           gmf-backgroundlayerselector-map="::ctrl.map">
         </gmf-backgroundlayerselector>
     </div>

--- a/contribs/gmf/examples/backgroundlayerselector.js
+++ b/contribs/gmf/examples/backgroundlayerselector.js
@@ -35,6 +35,12 @@ app.MainController = function(gmfThemes) {
   gmfThemes.loadThemes();
 
   /**
+   * @type {Object.<string, string>}
+   * @export
+   */
+  this.dimensions = {};
+
+  /**
    * @type {ol.Map}
    * @export
    */

--- a/contribs/gmf/src/directives/backgroundlayerselector.js
+++ b/contribs/gmf/src/directives/backgroundlayerselector.js
@@ -27,7 +27,8 @@ gmf.module.value('gmfBackgroundlayerselectorTemplateUrl',
  * Example:
  *
  *      <gmf-backgroundlayerselector
- *        gmf-backgroundlayerselector-map="::ctrl.map">
+ *        gmf-backgroundlayerselector-dimensions="::ctrl.dimensions"
+ *        gmf-backgroundlayerselector-map="::ctrl.map"
  *        gmf-backgroundlayerselector-select="onBackgroundSelected()">
  *      </gmf-backgroundlayerselector>
  *
@@ -35,6 +36,8 @@ gmf.module.value('gmfBackgroundlayerselectorTemplateUrl',
  *
  *  * thumbnail: The URL used for the icon.
  *
+ * @htmlAttribute {Object.<string, string>} gmf-backgroundlayerselector-dimensions
+ *     The dimensions.
  * @htmlAttribute {ol.Map=} gmf-backgroundlayerselector-map The map.
  * @htmlAttribute {Function} gmf-backgroundlayerselector-select Function called
  *     when a layer was selected by the user.
@@ -50,8 +53,8 @@ gmf.backgroundlayerselectorDirective = function(
   return {
     restrict: 'E',
     scope: {
-      'map': '=gmfBackgroundlayerselectorMap',
       'dimensions': '=gmfBackgroundlayerselectorDimensions',
+      'map': '=gmfBackgroundlayerselectorMap',
       'select': '&?gmfBackgroundlayerselectorSelect'
     },
     bindToController: true,
@@ -79,6 +82,14 @@ gmf.module.directive('gmfBackgroundlayerselector',
  */
 gmf.BackgroundlayerselectorController = function($scope, ngeoBackgroundLayerMgr,
     gmfThemes) {
+
+  /**
+   * @type {Object.<string, string>}
+   * @export
+   */
+  this.dimensions;
+
+  goog.asserts.assert(this.dimensions, 'The dimensions object is required');
 
   /**
    * @type {ol.Map}
@@ -119,14 +130,6 @@ gmf.BackgroundlayerselectorController = function($scope, ngeoBackgroundLayerMgr,
 
   this.listenerKeys_.push(ol.events.listen(gmfThemes,
     gmf.ThemesEventType.CHANGE, this.handleThemesChange_, this));
-
-  /**
-   * @type {Object.<string, string>}
-   * @export
-   */
-  this.dimensions;
-
-  goog.asserts.assert(this.dimensions, 'The dimensions object is required');
 
   gmfThemes.getBgLayers(this.dimensions).then(function(layers) {
     this.bgLayers = layers;

